module-system/sov-modules-stf-template/src/app_template.rs

```diff
@@ -0,0 +1,281 @@
+        &mut self,
+        batch_workspace: WorkingSet<C::Storage>,
+        batch: Batch,
+        batch_data_and_hash: &BatchDataAndHash,
+    ) -> Result<(WorkingSet<C::Storage>, Vec<TransactionAndRawHash<C>>)> {
+        // Run the stateless verification, since it is stateless we don't commit.
+        match verify_txs_stateless(batch.take_transactions()) {
+            Ok(txs) => Ok((batch_workspace, txs)),
+            Err(e) => {
+                // Revert on error
+                self.checkpoint = Some(batch_workspace.revert());
+                error!("Stateless verification error - the sequencer included a transaction which was known to be invalid. {}\n", e);
+                Err(ApplyBatchError::Slashed {
+                    hash: batch_data_and_hash.hash,
+                    reason: SlashingReason::StatelessVerificationFailed,
+                })
+            }
+        }
+    }
+
+    fn execute_txs(
+        &mut self,
+        mut batch_workspace: WorkingSet<C::Storage>,
+        txs: Vec<TransactionAndRawHash<C>>,
+        batch_data_and_hash: &BatchDataAndHash,
+    ) -> Result<(WorkingSet<C::Storage>, Vec<TransactionReceipt<TxEffect>>)> {
+        let mut tx_receipts = Vec::with_capacity(txs.len());
+
+        // Process transactions in a loop, commit changes after every step of the loop.
+        for tx in txs {
+            // Run the stateful verification, possibly modifies the state.
+            let sender_address = match self
+                .runtime
+                .pre_dispatch_tx_hook(tx.tx.clone(), &mut batch_workspace)
+            {
+                Ok(verified_tx) => verified_tx,
+                Err(e) => {
+                    // Don't revert any state changes made by the pre_dispatch_hook even if the Tx is rejected.
+                    // For example nonce for the relevant account is incremented.
+                    error!("Stateful verification error - the sequencer included an invalid transaction: {}", e);
+                    let receipt = TransactionReceipt {
+                        tx_hash: tx.raw_tx_hash,
+                        body_to_save: None,
+                        events: batch_workspace.take_events(),
+                        receipt: TxEffect::Reverted,
+                    };
+
+                    tx_receipts.push(receipt);
+                    continue;
+                }
+            };
+
+            match RT::decode_call(tx.tx.runtime_msg()) {
+                Ok(msg) => {
+                    let ctx = C::new(sender_address.clone());
+                    let tx_result = self.runtime.dispatch_call(msg, &mut batch_workspace, &ctx);
+
+                    self.runtime
+                        .post_dispatch_tx_hook(&tx.tx, &mut batch_workspace)
+                        .expect("Impossible happened: error in post_dispatch_tx_hook");
+
+                    let tx_effect = match tx_result {
+                        Ok(_) => TxEffect::Successful,
+                        Err(_e) => {
```

Comment:
Maybe let's log the error?

Reply:
I put `debug` it is ok for a single tx to fail here.